juju/unit.py

```diff
@@ -15,7 +15,9 @@
 
 
 class Unit(model.ModelEntity):
-    name: str
+    @property
+    def name(self) -> str:
+        return self.entity_id
```

Comment:
Was this a type annotation error? name never actually set outside our tests?

Reply:
There are, weirdly two sources of truth for the name:
- any ModelEntity, when created is pinned by entity id.
- then state updates are dispatched to safe data by entity id.
- safe_data["name"] exists and equals entity id because of the dispatch.
- __getattr__(self, "name") used to fetch it from safe_data

I'm basically making this a little more sane.
It's true that I didn't have to make this change in this very PR, but it helped me with types.... or tests for something.

Reply:
I have no objections to making the change, it just jumped out at me. Thanks for explaining